File: Sandbox/Scripts_CSharp/Source/Camera.cs
using System;

using Vanta;

namespace Sandbox {

    public class Camera : Entity {
        
        public float DistanceZ = 5.0f;

        private Entity Player;

        void OnCreate() {
            Position = new Vector3(Position.XY, DistanceZ);

            Player = GetEntityByName("Player");
            if (!Player) {
                Log.Error("Failed to find player entity!");
            }
        }

        void OnUpdate(float ts) {
            if (Player)
                Position = new Vector3(Player.Position.XY, DistanceZ);
        }
    }
}

File: Sandbox/Scripts/Native/Source/Camera.cpp
#include "Camera.hpp"

using namespace Vanta;

namespace Sandbox {

    void Camera::OnCreate() {
        m_Player = GetEntityByName("Player");
        if (!m_Player) {
            Vanta::Log::Error("Failed to find player entity!");
        }
    }

    void Camera::OnUpdate(double) {
        if (!m_Player)
            return;

        auto tr = GetComponent<TransformComponent>();
        auto& playerPos = m_Player.GetComponent<TransformComponent>().GetPosition();

        if (m_Player)
            tr.SetPosition({ playerPos.X, playerPos.Y, DistanceZ });
    }
}
